Modules/Authentication/RequireRoleAttribute.cs

```diff
@@ -0,0 +1,21 @@
+﻿using GenHTTP.Modules.Authentication.Roles;
+using GenHTTP.Modules.Reflection;
+
+namespace GenHTTP.Modules.Authentication;
+
+/// <summary>
+/// When annotated on a service method, requests will only be allowed
+/// if the authenticated user has the specified roles.
+/// </summary>
+/// <param name="roles"></param>
+[AttributeUsage(AttributeTargets.Method)]
+public class RequireRoleAttribute(params string[] roles) : InterceptWithAttribute<RoleInterceptor>
+{
+
+    /// <summary>
+    /// The roles which need to be present in order to let
+    /// the request pass.
+    /// </summary>
+    public string[] Roles { get; } = roles;

```

Comment:
I'd prefer to drop this field when you use primary ctor and to comply with naming conventions rename the parameter (generated property) in the primary ctor to "Roles" instead of "roles".  The documentation from the property could be moved to the ctor param.